refactor/refactor.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+// Package refactor implements different refactoring operations over Rego modules.
+package refactor
+
+import (
+	"fmt"
+
+	"github.com/open-policy-agent/opa/ast"
+)
+
+// Error defines the structure of errors returned by refactor.
+type Error struct {
+	Message  string        `json:"message"`
+	Location *ast.Location `json:"location,omitempty"`
+}
+
+func (e Error) Error() string {
+
+	if e.Location != nil {
+		return e.Location.Format(e.Message)
+	}
+
+	return e.Message
+}
+
+// Refactor implements different refactoring operations over Rego modules eg. renaming packages.
+type Refactor struct {
+}
+
+// New returns a new Refactor object.
+func New() *Refactor {
+	return &Refactor{}
+}
+
+// MoveQuery holds the set of Rego modules whose package paths and other references are to be rewritten
+// as per the mapping defined in SrcDstMapping.
+// If validate is true, the moved modules will be compiled to ensure they are valid.
+type MoveQuery struct {
+	Modules       map[string]*ast.Module
+	SrcDstMapping map[string]string
+	validate      bool
+}
+
+// WithValidation controls whether to compile moved modules to ensure they are valid.
+func (mq MoveQuery) WithValidation(v bool) MoveQuery {
+	mq.validate = v
+	return mq
+}
+
+// MoveQueryResult defines the output of a move query and holds the rewritten modules with updated packages paths
+// and references.
+type MoveQueryResult struct {
+	Result map[string]*ast.Module `json:"result"`
+}
+
+// validate validates moved modules by compiling them.
+func (mqr *MoveQueryResult) validate() error {
+	compiler := ast.NewCompiler()
+	compiler.Compile(mqr.Result)
+
+	if compiler.Failed() {
+		return compiler.Errors
+	}
+	return nil
+}
+
+// Move rewrites Rego code by updating package paths and other references in q's modules as per
+// the mapping specified in q.
+func (r *Refactor) Move(q MoveQuery) (*MoveQueryResult, error) {
+
+	for _, module := range q.Modules {
+		t := ast.NewGenericTransformer(func(x interface{}) (interface{}, error) {
+			if s, ok := x.(ast.Ref); ok {
+				// exact match
+				if val, found := q.SrcDstMapping[s.String()]; found {
```

Comment:
Relying on the String() representation of refs here is a bit brittle. I'd prefer if the implemented preprocessed the srcDst map by parsing the strings into refs and then relying on an ast.ValueMap/ast.Compare call here.

Reply:
Nit: do we actually need this case? It should be handled by the case below on L89 (I think).

Reply:
You're right. I'll create a PR shortly with this removed.